src/useRouteQueryParam/formats/RouteParam.ts

```diff
@@ -0,0 +1,105 @@
+import { LocationQueryValue } from 'vue-router'
+import { UseRouteQuery } from '@/useRouteQuery/useRouteQuery'
+import { InvalidRouteParamValue, isInvalidRouteParamValue, isNotInvalidRouteParamValue } from '@/useRouteQueryParam/formats/InvalidRouteParamValue'
+import { asArray } from '@/utilities/arrays'
+
+const IS_ROUTE_PARAM_SYMBOL: unique symbol = Symbol()
+
+export type RouteParamClassArgs<T> = {
+  key: string,
+  defaultValue: T | T[],
+}
+
+// adding any here so RouteParamClass can be used without passing a generic
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type RouteParamClass<T = any> = new (args: RouteParamClassArgs<T>) => RouteParam<T>
```

Comment:
do you know why we can't use `InstanceType<RouteParam<T>>`?

Reply:
I tried a few different things here and this was what I could get to work. But `InstanceType` is just that. Its the instance and we want a constructor here. 

Might not be what you're asking. So leaving this unresolved.

Reply:
Maybe idk typescript but I thought `function(input:SomeClass)` is a function that takes an instance of `SomeClass` as it's only argument. I would expect `InstanceType` to get me the underlying type for this instance. But it doesn't work here, so clearly I'm mistaken

Reply:
You're explaining the types correctly. But what you're missing is the functions in this PR don't accept a `RouteParam` (instance) but a `RouteParamClass` (constructor). 

<img width="565" alt="image" src="https://user-images.githubusercontent.com/6200442/210424773-ca5824ff-ebe0-49a6-92d2-14127b6c7533.png">

Reply:
right, my original comment was that `RouteParamClass` seemed like an unnecessary abstraction. I expected you to be able to use `InstaceType<RouteParam<T>>`

Reply:
Maybe the definition for `InstanceType` will help explain why its not what you think it is 

```typescript
type InstanceType<T extends abstract new (...args: any) => any> = T extends abstract new (...args: any) => infer R ? R : any;
```

Reply:
As far as I know (or have been able to find) there isn't a way to go from an instance (RouteParam) to its constructor type.